### ISSUE TITLE
The workspaces don't work with testfx

### ISSUE THREAD

**Issue by chinzen** [edvin/tornadofx#500]
When I try to utilize the workspaces, testfx breaks in a curious manner.

Let's say I have an app
```kotlin
class MyApplication : App(Workspace::class) {
    override fun onBeforeShow(view: UIComponent) {
        workspace.dock<MainView>()
    }
}

class MainView : View() {
    override val root = button("name") {
        id = "button"
    }
}
```

And a test case for it
```kotlin
import org.testfx.framework.junit.ApplicationTest

class MyApplicationTest : ApplicationTest() {
    lateinit var app: MyApplication
    
    override fun start(stage: Stage) {
        app = MyApplication()
        app.start(stage)
    }
    
    override fun stop() {
        app.stop()
    }
    
    @Test
    fun `test 1`() {
        clickOn("#button")
    }
    
    @Test
    fun `test 2`() {
        clickOn("#button")
    }
}
```
The `test 1` works just fine and looks like that
![workspace-with-view](https://user-images.githubusercontent.com/32562821/31704857-cb3855f0-b3eb-11e7-87a6-f6c5b52021ec.png)

But the `test 2` always fails. More precisely, it seems that it's unable to dock the view into the workspace, resulting in the following picture
![workspace-without-view](https://user-images.githubusercontent.com/32562821/31704883-ea897b32-b3eb-11e7-8d0a-3846f21e7eed.png)

Thus the testfx robot throws `the query "#button" returned no nodes`.

I wonder if I'm missing something...

**Comment by edvin** [comment]
`ApplicationTest` will start multiple applications when you add multiple test functions like this, so I'm guessing the primary stage gets lost somehow. I'm not sure how `ApplicationTest` is supposed to work, but I'm guessing you need to do some house keeping before/after each test invocation. @hastebrot might have some insight? :)

**Comment by chinzen** [comment]
Thank you for the hint.

I tried to track the primary stage, but to no avail... The stage doesn't seem to change, at least its JVM object id is stable throughout the whole program.

Also testfx, as far as I was able to understand, instantiates the stage only once. In a sequence digram its `registerPrimaryStage` method looks somehow like this:
![register-primary-stage](https://user-images.githubusercontent.com/32562821/31851514-ee8c3e00-b66f-11e7-8948-6a6e3e0bf635.png)
And by `[primary_stage_future is not done] launch(app_class)` I meant that it launches the app only if the `primary_stage_future` is empty.

I also tried to rewrite the test case into the form where it create only one app:
```kotlin
class MyTestCase : ApplicationTest() {
    companion object {
        lateinit var app: MyApplication

        @JvmStatic @BeforeClass
        fun setup() {
            app = MyApplication()
        }
    }

    override fun start(stage: Stage) {
        app.start(stage)
    }
    
    override fun stop() {
        app.stop()
    }
```
But that doesn't help either.

I'm afraid I will stop at that. At that point it seems more reasonable to replace the workspaces with a primitive history service, than try to debug it any further.

I also find it unusual that a gui framework doesn't have an official tutorial for gui testing. I bet many people would really appreciate that.
